### DISCUSSION TITLE
How to get client real ip？

### DISCUSSION THREAD

**Discussion by lq9958** [nuxt/nuxt#25059] [Questions]
Firstly, my API server aims to record the IP addresses of visitors for data analytics. 

I've attempted to use `useRequestHeaders` to retrieve values for `x-forwarded-for`, `x-real-ip`, and `user-agent`. However, the returned values only include `x-forwarded-for` and `user-agent`, with `x-forwarded-for` consistently being `127.0.0.1`. 

I am uncertain whether `useRequestHeaders` fetches data from the browser to the SSR server or from the SSR server to my API server. 
I've also tried using `getRequestIP` in the `server/middleware`, but the result remains `127.0.0.1`. 

It seems that there might be relevant configurations for SSR server forwarding that I haven't come across.

Currently, both the SSR server and API server are deployed locally on my machine. When I access my website from other PCs in the local network, the IP address obtained by the API server still reflects my local machine's IP address.

**Comment by lq9958** [comment]
I can only inject the client's IP directly by modifying the source code of `h3`, is there have some simple way ?
```javascript
async function proxyRequest(event, target, opts = {}) {
  let body;
  let duplex;
  if (PayloadMethods.has(event.method)) {
    if (opts.streamRequest) {
      body = getRequestWebStream(event);
      duplex = "half";
    } else {
      body = await readRawBody(event, false).catch(() => void 0);
    }
  }
  const method = opts.fetchOptions?.method || event.method;
  const clientAddress = event.node.req.socket.remoteAddress
  let clientRealIp = ''
  if (clientAddress === '::1') {
    clientRealIp = '127.0.0.1'
  } else {
    const reg = /^::ffff:(.*)/
    clientRealIp = clientAddress.match(reg)[1]
  }
  console.log(clientRealIp)
  const fetchHeaders = mergeHeaders(
    getProxyRequestHeaders(event),
    opts.fetchOptions?.headers,
    opts.headers, {
    'x-real-ip': clientRealIp
  }
  );
  return sendProxy(event, target, {
    ...opts,
    fetchOptions: {
      method,
      body,
      duplex,
      ...opts.fetchOptions,
      headers: fetchHeaders
    }
  });
}
```

**Comment by Mobbbb** [comment]
I use the nginx server to get the user's real IP address.
```nginx
location /api {
        proxy_pass http://127.0.0.1:3001/;
        proxy_set_header HOST $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
```
And then, I can get `IP`, `referer` from my api, like
```javascript
const headers = ctx.request.header
const ip = headers['x-forwarded-for']
const referer = headers.referer
```
